clients/src/test/java/org/apache/kafka/common/security/authenticator/ClientAuthenticationFailureTest.java

```diff
@@ -34,26 +34,48 @@
 import org.apache.kafka.common.security.auth.SecurityProtocol;
 import org.apache.kafka.common.serialization.StringDeserializer;
 import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.common.utils.Time;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import java.time.Duration;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
+@RunWith(value = Parameterized.class)
 public class ClientAuthenticationFailureTest {
 
     private NioEchoServer server;
     private Map<String, Object> saslServerConfigs;
     private Map<String, Object> saslClientConfigs;
     private final String topic = "test";
     private TestJaasConfig testJaasConfig;
+    private final int failedAuthenticationDelayMs;
+    private long startTimeMs;
+
+    public ClientAuthenticationFailureTest(int failedAuthenticationDelayMs) {
+        this.failedAuthenticationDelayMs = failedAuthenticationDelayMs;
+    }
+
+    @Parameterized.Parameters(name = "failedAuthenticationDelayMs={0}")
+    public static Collection<Object[]> data() {
+        List<Object[]> values = new ArrayList<>();
+        values.add(new Object[]{-1});
+        values.add(new Object[]{0});
+        values.add(new Object[]{5000});
```

Comment:
Doesn't this delay every test by 5 seconds? Needs to be a lot lower, just to cause the code to take the delay path (1ms perhaps?). We could have one test that delays slightly longer to test that it was actually delayed (don't think we are testing that here).

Reply:
We are testing if we actually delayed failed authentication response - we have an assert in `teardown()` to make sure we spent at least the configured amount of time.

Reply:
Sorry, I had missed that. Perhaps 500ms should be sufficient? Also, not sure if we need to delay so many tests (even if the delay is 500ms).